Add destinations api client (#1175)
planetlabs/planet-client-python: docs/cli/cli-destinations.md

---
title: CLI for Destinations API Tutorial
---

## Introduction

> The `planet destinations` command enables interaction with the [Destinations API](https://docs.planet.com/develop/apis/destinations/), which enables the creation, listing, and modifying of destinations as well as using destinations in other services to streamline data delivery. This tutorial takes you through the main commands available in the CLI.

Reviewer: ```suggestion
The `planet destinations` command provides an interface for creating, listing, and modifying destinations in the [Planet Destinations API](https://docs.planet.com/develop/apis/destinations/). This tutorial takes you through the main commands available in the CLI.
```

I removed `...as well as using destinations in other services to streamline data delivery...` because the wording was kind of confusing to me.  Does the `planet destinations` command actually provide a code path to use destinations in other services?  Or does the `planet destinations` command provide ways to get destination refs (strings) that users can include in requests to other services?

Author: I like your suggestion here. It is the latter.

## Core Workflows

### Create a Destination
To discover the supported cloud destinations, run the command `planet destinations create --help`. Once you have chosen your target cloud destination type, run the command `planet destinations create <type> --help` to discover the required and supported parameters (eg: `planet destinations create s3 --help`).

Finally, submit the full request:
```sh
planet destinations create s3 my-bucket us-west-2 AKIA... SECRET... --name my-s3-destination
```

> The newly created destination will be printed to standard out, with the destination reference under the key `pl:ref`, which can subsequently be used in Orders API and Subscriptions API requests as the delivery destination.

Reviewer: ```suggestion
The newly created destination will be printed to stdout, with the destination reference under the key `pl:ref`, which can subsequently be used in Orders API and Subscriptions API requests as the delivery destination.
```

### List Destinations
Listing destinations can be accomplished with the command `planet destinations list`. This will return all destinations within your organization.

You can get nicer formatting with `--pretty` or pipe it into `jq`, just like the other Planet CLIs.

#### Filtering
The `list` command supports filtering on a variety of fields. You can discover all filterable fields by running the command `planet destinations list --help`.

* `--archived / --is-not-archived`: Filter by archive status. Use --archived to include only archived destinations, or --is-not-archived to list only active (not archived) destinations.
* `--is-owner / --is-not-owner`: Filter by ownership. Use --is-owner to include only destinations owned by the requesting user, or --is-not-owner to include destinations not owned by the user.
* `--can-write / --can-not-write`: Filter by write access. Use --can-write to include only destinations the user can modify, or --can-not-write to list destinations with read-only access for the user.

> For example, to list destinations that are not archived and you can modify you would issue the following command:

Reviewer: Sounds good!

```sh
planet destinations list --not-archived --can-write
```

### Update Destinations

> The CLI conveniently moves all update actions to first class commands on the destination. The allowed update actions are archiving, unarchiving, renaming, and updating credentials. To discover all update actions run `planet destinations --help`.

Reviewer: ```suggestion
The CLI conveniently moves all update actions to first class commands on the destination. The supported actions are archive, unarchive, rename, and update credentials. To discover all update actions run `planet destinations --help`.
```
